### PR TITLE
generalized neuroscope waveform interface; updated Grosmark to v0.6.3

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
@bendichter 

1) updated all of Grosmark converter and interface codes for nwb-conversion-tools v0.6.3
2) generalized the waveform interface for all neuroscope files (.spk format); previously this was a part of experiment specific LFP (which is also now generalized per nwb-conversion-tools)
3) updated device description and LFP conversion factor to Ampliplex
4) generally refactored for improved performance and code reduction

### CODE REVIEW

File: buzsaki_lab_to_nwb/grosmark_code/grosmarkbehaviordatainterface.py [hunk 1]
```diff
@@ -17,42 +17,38 @@ class GrosmarkBehaviorInterface(BaseDataInterface):
...
-            check_module(nwbfile, 'behavior', 'contains behavioral data').add_data_interface(table)
+        matin = loadmat(str(sleep_state_fpath))['SleepState']['ints'][0][0]
+        table = TimeIntervals(name='states', description="Sleep states of animal.")
+        table.add_column(name='label', description="Sleep state.")
+        data = []
+        for name in matin.dtype.names:
+            for row in matin[name][0][0]:
+                data.append(dict(start_time=row[0], stop_time=row[1], label=state_label_names[name]))
+        [table.add_row(**row) for row in sorted(data, key=lambda x: x['start_time'])]
+        check_module(nwbfile, 'behavior', "Contains behavioral data.").add_data_interface(table)
```
Comment (bendichter) [inline buzsaki_lab_to_nwb/grosmark_code/grosmarkbehaviordatainterface.py:44]:
```suggestion
        check_module(nwbfile, 'behavior', "Contains behavioral data.").add(table)
```

File: buzsaki_lab_to_nwb/neuroscopewaveformdatainterface.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        all_shank_channels.sort()
+        spikes_nsamples = int(root.find('neuroscope').find('spikes').find('nSamples').text)
+
+        write_spike_waveforms(
+            nwbfile,
+            session_path,
+            spikes_nsamples=spikes_nsamples,
+            shank_channels=shank_channels,
+            stub_test=stub_test
+        )
```
Comment (bendichter) [inline buzsaki_lab_to_nwb/neuroscopewaveformdatainterface.py:43]: Might this be better as an optional argument for the `NeuroscopeRecordingInterface`?

**Review by bendichter — COMMENTED** [review]
see above

File: buzsaki_lab_to_nwb/neuroscopewaveformdatainterface.py (hunk 2 above)
Comment (CodyCBakerPhD) [reply buzsaki_lab_to_nwb/neuroscopewaveformdatainterface.py:43]: Well, it really ought to be included and written on the `SortingExtractor` side if I'm honest, but that change haven't coordinated those changes through on the various repo's yet; I do have another idea, though - maybe we could add an option to the `NwbSortingExtractor.write_sorting()` to write waveforms as `EventSeries` or as jagged unit table columns (and raise `NotImplemented` until all the pieces are in place for that on `pynwb` and `nwbwidgets`)

**Comment by CodyCBakerPhD** [comment]
With the upcoming changes to pynwb and spikeextractors waveforms (including neuroscope automatically writing them), this needs to be changed to reflect that. Small changes though, won't take long

**Comment by CodyCBakerPhD** [comment]
@bendichter This PR is ready for final review now.

**Comment by CodyCBakerPhD** [comment]
Refactoring/cleaning repo to newest version

**Pull request closed without merging**